ast/src/parsed/mod.rs

```diff
@@ -650,25 +613,42 @@ pub trait SourceReference {
 }
 
 macro_rules! impl_source_reference {
+    // Version for types with generic parameter
+    ($enum:ident<$generic:ident>, $($variant:ident),*) => {
+        impl<$generic> SourceReference for $enum<$generic> {
+            fn source_reference(&self) -> &SourceRef {
+                match self {
+                    $( $enum::$variant(src, ..) => src, )*
+                }
+            }
+
+            fn source_reference_mut(&mut self) -> &mut SourceRef {
+                match self {
+                    $( $enum::$variant(src, ..) => src, )*
+                }
+            }
+        }
+    };
+    // Version for types without generic parameter
     ($enum:ident, $($variant:ident),*) => {
-        impl<E> SourceReference for $enum<E> {
+        impl SourceReference for $enum {
             fn source_reference(&self) -> &SourceRef {
                 match self {
-                    $( $enum::$variant(src, _) => src, )*
+                    $( $enum::$variant(src, ..) => src, )*
                 }
             }
 
             fn source_reference_mut(&mut self) -> &mut SourceRef {
                 match self {
-                    $( $enum::$variant(src, _) => src, )*
+                    $( $enum::$variant(src, ..) => src, )*
                 }
             }
         }
-    }
+    };
 }
 
 impl_source_reference!(
-    Expression,
+    Expression<E>,
```

Comment:
Why is expression also mentioned below?

Reply:
The expression below is the Expression case inside the PilStatement enum.